hal/sparcv8leon3/gaisler/console.c

```diff
@@ -109,13 +109,21 @@ void hal_consolePrint(int attr, const char *s)
 
 void _hal_consoleInit(void)
 {
+	char c;
 	gaisler_setIomuxCfg(CONSOLE_TX, 0x1, 0, 0);
 	gaisler_setIomuxCfg(CONSOLE_RX, 0x1, 0, 0);
 #ifdef __CPU_GR716
 	_gr716_cguClkEnable(cgu_primary, CONSOLE_CGU);
 #endif
 	halconsole_common.uart = VADDR_CONSOLE;
-	*(halconsole_common.uart + uart_ctrl) = TX_EN;
+	*(halconsole_common.uart + uart_ctrl) = 0;
+
+	/* Clear UART FIFO */
+	while ((*(halconsole_common.uart + uart_status) & (1 << 0)) != 0) {
+		c = *(halconsole_common.uart + uart_data) & 0xff;
```

Comment:
```suggestion
		(void)*(halconsole_common.uart + uart_data);
```
Should be enough to cause access on volatile stuff